### ISSUE TITLE
Custom build flags choke on prefix workspaces

### ISSUE THREAD

**Issue by ulysses4ever** [bazelbuild/bazel#9177]
### Description of the problem

I believe custom build flags don't play well with proper `@namespace//...`-like references.

### Bugs: what's the simplest, easiest way to reproduce this bug?

I set up [a repository with a repro](https://github.com/ulysses4ever/bazel-user-settings-example/) -- check out the README. It follows closely the running example of drinks and flavors [in the documentation](https://docs.bazel.build/versions/0.28.0/skylark/config.html). Except that it actually sets up a named workspace. 

In the `master` branch of the repo, everything is fine: I'm able to pass the flag in the command line. In the [`bug`](https://github.com/ulysses4ever/bazel-user-settings-example/commit/1d79b746b0323e0450a99aedea9e8e3c3d924c07) branch I add the workspace prefix when referencing the flag target from the attribute of my rule. This breaks the setup: I pass the flag in the command-line as before but it has no effect. If I try to add the workspace prefix when referencing the flag from the command-line, I get `"Unrecognized option"`.

### What operating system are you running Bazel on?

Ubuntu 18.04

### What's the output of `bazel info release`?

```
release 0.28.0- (@non-git)
```

### If `bazel info release` returns "development version" or "(@non-git)", tell us how you built Bazel.

This is the version of Bazel imported by Nixpkgs version referenced in the current [`rules_haskell`](https://github.com/tweag/rules_haskell/) (master branch).

###  Have you found anything relevant by searching the web?

No. I only looked up the Issues here.

**Comment by ulysses4ever** [comment]
May I ping @juliexxia who did most of the job of implementing this exciting feature!

**Comment by gregestren** [comment]
This is a cutting edge feature so I'm not surprised this part doesn't work yet. I don't think anyone can take this on right now, but please re-ping whenever you feel to help us figure out a practical fix timeline.

Appreciate your patient guinea pigging!

**Comment by aherrmann** [comment]
I've tested this issue again on the more recent Bazel version 2.1. The original issue is still present. I.e. defining the workspace name prefix on the flag causes the flag to be ignored, if it is defined locally. Meaning, this issue is still open.

However, what is working now is to prefix the workspace name if the flag is defined in an external workspace. This is the use-case that is interesting for rule authors, because it allows them to define configurations in their rules sets that users can then affect on the command-line.

I've forked and updated @ulysses4ever's reproduction [here](https://github.com/aherrmann/bazel-user-settings-example). This illustrates that if a rules set, say `rules_example`, defines a flag `//:favorite_flavor`, then users can change it in their project that uses `rules_example` as an external repository using the command-line flag like so:
```
bazel build //:my_drink --@rules_example//:favorite_flavor=True
```
